packit/command_handler.py

```diff
@@ -157,3 +166,14 @@ def run_command(
         if return_output:
             return out
         return None
+
+    def clean(self):
+        if self.is_running:
+            logger.info("Deleting sandcastle pod.")
+            self.sandcastle.delete_pod()
+            self.is_running = False
+        else:
+            logger.info("Sandcastle pod is not running, nothing to clean up")
+
+    def __del__(self):
+        self.clean()
```

Comment:
Just thinking out loud. Is this enough for resolving all bad situations in a clean way?

Will this run in case of error during pod preparing or during any packit workflow?

But TBH, I can' think of any better approach. Maybe some context manager (e.g.  `__enter__`/`__exit__` methods)?

Reply:
This is a good point. From the [Python docs](https://docs.python.org/3/reference/datamodel.html):

> It is not guaranteed that __del__() methods are called for objects that still exist when the interpreter exits.

So yes, probably to have a better control over the lifecycle of the sandcastle pod a context manager should be used here.

Not sure if this should be part of this PR, though.

Reply:
> This is a good point. From the [Python docs](https://docs.python.org/3/reference/datamodel.html):
> 
> > It is not guaranteed that `__del__()` methods are called for objects that still exist when the interpreter exits.
> 
> So yes, probably to have a better control over the lifecycle of the sandcastle pod a context manager should be used here.
> 
> Not sure if this should be part of this PR, though.

BTW I suppose to let it there, as last chance for cleanup, but as mentioned __del__ could be committed in some cases. We had it also inside `requre` and there were anyther issue, that I think `self` object what little bit mismatched, so that I had to have there some check or maybe some try except.

Reply:
https://github.com/packit/packit-service/issues/1009